### DISCUSSION TITLE
How to inspect the sandbox for cram tests

### DISCUSSION THREAD

**Discussion by Blaisorblade** [ocaml/dune#11167] [General]
Cram tests run into a temporary sandbox folder that gets deleted — is there a way to prevent deletion, to allow inspecting the sandbox or even modifying it (when debugging the test)?

In my case, my cram tests generates some file `foo.v`, and then compiles it with Coq. I'm getting unexpected errors so I'd like to inspect `foo.v` — the only way I found is to add `cat foo.v` to my cram test, use promotion to update `run.t`, then copy-paste the output out of the `.t`. But this seems hacky, and would get worse if I had multiple generated files, or some of them were binary (and yes, something like base64 encoding could make the hack applicable there too).

Or I can extract the commands from the `run.t` file and run them outside — again, a hassle if there are many commands.

All this gets worse if the problem isn't repeatable — but I wouldn't restrict a solution to failing tests: I might want to inspect the sandbox even when the test succeeds.

EDIT: I already tried `--sandbox=none` — but I can't find outputs, and if `run.t` needs changes, the diff shows that the test still runs under `_build/.sandbox`. Example:
```
dune b --sandbox=none @test_float_lit
Entering directory '/Users/pgiarrusso1/git-bedrock/bhv'
 llvm-config fmdeps/cpp2v-core/rocq-bluerock-cpp2v/build/llvm-lib-path.txt
         sed fmdeps/cpp2v-core/rocq-bluerock-cpp2v/build/llvm-lib-path.txt
File "fmdeps/cpp2v-core/rocq-bluerock-brick/tests/cpp2v/test_float_lit.t/run.t", line 1, characters 0-0:
diff --git a/_build/.sandbox/17a92c64d74332be4f62d9b0d978432b/default/fmdeps/cpp2v-core/rocq-bluerock-brick/tests/cpp2v/test_float_lit.t/run.t b/_build/.sandbox/17a92c64d74332be4f62d9b0d978432b/default/fmdeps/cpp2v-core/rocq-bluerock-brick/tests/cpp2v/test_float_lit.t/run.t.corrected
index 7aa49eecb5..0ad369b408 100644
--- a/_build/.sandbox/17a92c64d74332be4f62d9b0d978432b/default/fmdeps/cpp2v-core/rocq-bluerock-brick/tests/cpp2v/test_float_lit.t/run.t
+++ b/_build/.sandbox/17a92c64d74332be4f62d9b0d978432b/default/fmdeps/cpp2v-core/rocq-bluerock-brick/tests/cpp2v/test_float_lit.t/run.t.corrected
[...]
```

**Comment by rgrinberg** [comment]
@Alizter I think you've added a setting to keep a sandbox around if a rule fails? Shall we bring that back perhaps. I think we can just go ahead with something like that except instead of keeping the sandbox spot, we can move it into some safe space for such broken sandboxes.
